src/main/java/frc/robot/subsystems/leds/LEDStrip.java

```diff
@@ -26,4 +36,51 @@ public void setSolidColor(Color color) {
         }
         led.setData(ledBuffer);
     }
+
+    //increment the colors in the cont buffer, does not set the leds
+    public void updateContinuousColor(Color color){
+        int inc = (int) Math.ceil(ledTimer.get() * LEDS_PER_SEC);
+        inc = Math.max(1, inc);
+        for (int i = contLedBuffer.getLength() - inc; i > 0; i--){
+            contLedBuffer.setLED(i, contLedBuffer.getLED(i - inc));
+        }
+        for(int i = 0; i < inc; i++){
+            contLedBuffer.setLED(i, color);
+        }
+        ledTimer.reset();
+        ledTimer.start();
+        
+    }
+
+    //set cont buffer on to the leds
+    public void setContinuousColor(){
```

Comment:
Is there a practical reason why the `setData(buffer)` call for continuous is separate from the "update buffer" call, and why the continuous buffer is separate from the solid buffer? It seems like right now every call to a "update continuous buffer" method is immediately proceeded by a "set buffer" call, and I think you can use the same buffer for both continuous mode and solid color since one simply overrides the other.

Reply:
yes, often you want to set the buffer without moving it up.